.github/workflows/cloud-run-tests.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Run tests
+
+on: push
+
```

Comment:
Looks great! I think it is good to have tests run on each push. I think if you expect a moderate level pf false failures (i.e. caused by a connection issue or outage in one of the services you are accession) then you may also want to add the option of triggering a test. See [link](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/).

I think testing on push is important as that way you can't forget to run your test. You can batch your pushes to origin if you are worried about the test overhead.

Reply:
Hi Rose, thanks for the input. I am not currently expecting false failures but if they are happening I will implement that